pkg/daemons/agent/agent_windows.go

```diff
@@ -2,5 +2,171 @@
+		argsMap["cni-conf-dir"] = cfg.CNIConfDir
+	}
+	if cfg.CNIBinDir != "" {
+		argsMap["cni-bin-dir"] = cfg.CNIBinDir
+	}
+	if cfg.CNIPlugin {
+		argsMap["network-plugin"] = "cni"
+	}
+	if len(cfg.ClusterDNS) > 0 {
+		argsMap["cluster-dns"] = util.JoinIPs(cfg.ClusterDNSs)
+	}
+	if cfg.ResolvConf != "" {
+		argsMap["resolv-conf"] = cfg.ResolvConf
+	}
+	if cfg.RuntimeSocket != "" {
+		argsMap["container-runtime"] = "remote"
+		argsMap["serialize-image-pulls"] = "false"
+		checkRuntimeEndpoint(cfg, argsMap)
+	} else if cfg.PauseImage != "" {
+		argsMap["pod-infra-container-image"] = cfg.PauseImage
+	}
+	if cfg.ListenAddress != "" {
+		argsMap["address"] = cfg.ListenAddress
+	}
+	if cfg.ClientCA != "" {
+		argsMap["anonymous-auth"] = "false"
+		argsMap["client-ca-file"] = cfg.ClientCA
+	}
+	if cfg.ServingKubeletCert != "" && cfg.ServingKubeletKey != "" {
+		argsMap["tls-cert-file"] = cfg.ServingKubeletCert
+		argsMap["tls-private-key-file"] = cfg.ServingKubeletKey
+	}
+	if cfg.NodeName != "" {
+		argsMap["hostname-override"] = cfg.NodeName
+	}
+	defaultIP, err := net.ChooseHostInterface()
+	if err != nil || defaultIP.String() != cfg.NodeIP {
+		argsMap["node-ip"] = cfg.NodeIP
+	}
+	kubeletRoot, runtimeRoot, hasCFS, hasPIDs := cgroups.CheckCgroups()
+	if !hasCFS {
+		logrus.Warn("Disabling CPU quotas due to missing cpu.cfs_period_us")
+		argsMap["cpu-cfs-quota"] = "false"
+	}
+	if !hasPIDs {
+		logrus.Warn("Disabling pod PIDs limit feature due to missing cgroup pids support")
+		argsMap["cgroups-per-qos"] = "false"
+		argsMap["enforce-node-allocatable"] = ""
+	}
+	if kubeletRoot != "" {
+		argsMap["kubelet-cgroups"] = kubeletRoot
+	}
+	if runtimeRoot != "" {
+		argsMap["runtime-cgroups"] = runtimeRoot
+	}
+	if system.RunningInUserNS() {
+		argsMap["feature-gates"] = addFeatureGate(argsMap["feature-gates"], "DevicePlugins=false")
+	}
+
+	argsMap["node-labels"] = strings.Join(cfg.NodeLabels, ",")
+	if len(cfg.NodeTaints) > 0 {
+		argsMap["register-with-taints"] = strings.Join(cfg.NodeTaints, ",")
+	}
+	if !cfg.DisableCCM {
+		argsMap["cloud-provider"] = "external"
+	}
+
+	if ImageCredProvAvailable(cfg) {
+		logrus.Infof("Kubelet image credential provider bin dir and configuration file found.")
+		argsMap["feature-gates"] = addFeatureGate(argsMap["feature-gates"], "KubeletCredentialProviders=true")
+		argsMap["image-credential-provider-bin-dir"] = cfg.ImageCredProvBinDir
+		argsMap["image-credential-provider-config"] = cfg.ImageCredProvConfig
+	}
+
+	if cfg.Rootless {
+		createRootlessConfig(argsMap, hasCFS, hasCFS)
+	}
+
+	if cfg.ProtectKernelDefaults {
+		argsMap["protect-kernel-defaults"] = "true"
+	}
+	return argsMap
+}
+
+func waitForManagementIp(networkName string) string {
+	for range time.Tick(time.Second * 5) {
+		network, err := hcsshim.GetHNSEndpointByName(networkName)
+		if err != nil {
+			logrus.WithError(err).Warning("can't find %s, retrying", networkName)
+			continue
+		}
+		return network.IPAddress.String()
+	}
+	return ""
```

Comment:
When is this case encountered?

Reply:
This is used for determining the source vip for kube proxy only on Windows.